presto-hive/src/main/java/com/facebook/presto/hive/authentication/KerberosHiveMetastoreAuthentication.java

```diff
@@ -56,6 +68,89 @@ public KerberosHiveMetastoreAuthentication(String hiveMetastoreServicePrincipal,
         this.hdfsWireEncryptionEnabled = hdfsWireEncryptionEnabled;
     }
 
+    @Override
+    public TTransport authenticateWithToken(TTransport rawTransport, String tokenStrForm)
+    {
+        try {
+            Token<DelegationTokenIdentifier> t = new Token<DelegationTokenIdentifier>();
+            t.decodeFromUrlString(tokenStrForm);
+
+            Map<String, String> saslProps = ImmutableMap.of(
+                    Sasl.QOP, "auth",
+                    Sasl.SERVER_AUTH, "true");
+
+            TTransport saslTransport = new TSaslClientTransport(
+                    "DIGEST-MD5",
```

Comment:
It's might be out of the scope of this pr.   But any chance to make it krb5/gss-api compatible? https://tools.ietf.org/html/rfc1964 The Kerberos Version 5 GSS-API Mechanism

Reply:
Not sure I got fully understood your question, Actually all these functions coming as the same implementation of hive (talking to secure HMS over TSaslTransport),